### ISSUE TITLE
♻️ Unified experiences table a bit painful

### ISSUE THREAD

**Issue by petertgiles** [GCTC-NTGC/gc-digital-talent#12279]
## ♻️ Debt/Refactor

A while ago, we refactored our experiences by [putting them in one table](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9899).  This worked well for a while but now that work experiences has its own relationships, it's becoming a bit painful.

## 🕵️ Details

Pros of the current setup:
- Can query a single table to get all experiences
- Can guarantee RI and avoid polymorphism for relationships to experiences like experience_skill and pool_candidate_education_requirement_experience

Cons of current setup:
- Can't guarantee RI for relationships inside experiences, like work_experience.department
- Can't use Lighthouse connect/disconnect for relationships inside experiences, like work_experience.department
- All the makeJsonProperty*Attribute properties are complicated
- Fields are harder to manually query and view in the database

## 🙋‍♀️ Proposed Solution

Let's go back to separate tables.
## ✅ Acceptance Criteria

- [ ] experiences handling is less painful, especially relationships inside work experience

## 🛑 Blockers

```[tasklist]
### Blocked By
```

**Comment by tristan-orourke** [comment]
Considering alternatives:
- We could keep them all in one table, split all json fields into real columns, and the table becomes wide and sparse.
- We could go back to 5 different tables, some relationships become polymorphic
- We could have one table of experiences, with five tables of experience details, and then use one or the other for relationships. The relation between the base experience and its details is a polymorphic one-to-one relationship.
- Rip out work experiences into its own because its too complicated. Other experiences stay as they are.

**Comment by tristan-orourke** [comment]
Is there really much advantage in being able to query for all experiences at once? Data wise, each experience type is actually quite different. All they all really share, organization and dates, and skill relationships, and education relationships.

If its for UI (eg Career Timeline page) its not very hard to query for the five types of experiences.

QUESTION: if we split this up in the database, can we split this up more in the schema?

**Comment by tristan-orourke** [comment]
Leaning towards going through with us, using option 2 (what we had before), but the question of whether to prioritize this or put it off until after Talent Management milestones are done is undecided.
